src/Docker/Client/Types.hs

```diff
@@ -1391,7 +1392,9 @@ instance FromJSON ContainerConfig where
         image <- o .: "Image"
         volumes <- o .: "Volumes"
         workingDir <- o .:? "WorkingDir"
-        entrypoint <- o .:? "Entrypoint"
+        entrypoint <- asum [ o .:? "Entrypoint" .!= []
+                           , (:[]) <$> o .: "Entrypoint"
```

Comment:
@kseo Adding tests would be great. Thanks! I've pushed changes to my dev branch that fix Travis tests and adds a matrix build for GHC 8.0 so if you could rebase you're changes that would be great.